suitcase/mongo_embedded/__init__.py

```diff
@@ -0,0 +1,495 @@
+                      for descriptor, eventpage in event_buffer.items()]
+        self._volatile_db.event.bulk_write(operations, ordered=False)
+
+    def _updateone_eventpage(self, descriptor_id, event_page):
+        """
+        Creates the UpdateOne command that gets used with bulk_write.
+        """
+        event_size = sys.getsizeof(event_page)
+
+        data_string = {'data.' + key: {'$each': value_array}
+                       for key, value_array in event_page['data'].items()}
+
+        timestamp_string = {'timestamps.' + key: {'$each': value_array}
+                            for key, value_array
+                            in event_page['timestamps'].items()}
+
+        filled_string = {'filled.' + key: {'$each': value_array}
+                         for key, value_array in event_page['filled'].items()}
+
+        update_string = {**data_string, **timestamp_string, **filled_string}
+
+        count = len(event_page['seq_num'])
+        self._event_count[descriptor_id] += count
+
+        return UpdateOne(
+            {'descriptor': descriptor_id, 'size': {'$lt': self._PAGE_SIZE}},
+            {'$push': {'uid': {'$each': event_page['uid']},
+                       'time': {'$each': event_page['time']},
+                       'seq_num': {'$each': event_page['seq_num']},
+                       **update_string},
+             '$inc': {'size': event_size},
+             '$min': {'first_index': self._event_count[descriptor_id] - count},
+             '$max': {'last_index': self._event_count[descriptor_id]-1}},
+            upsert=True)
+
+    def _updateone_datumpage(self, resource_id, datum_page):
+        """
+        Creates the UpdateOne command that gets used with bulk_write.
+        """
+        datum_size = sys.getsizeof(datum_page)
+
+        kwargs_string = {'datum_kwargs.' + key: {'$each': value_array}
+                         for key, value_array
+                         in datum_page['datum_kwargs'].items()}
+
+        count = len(datum_page['seq_number'])
+        self._datum_count[resource_id] += count
+
+        return UpdateOne(
+            {'resource': resource_id, 'size': {'$lt': self._PAGE_SIZE}},
+            {'$push': {'datum_id': {'$each': datum_page['uid']},
+                       **kwargs_string},
+             '$inc': {'size': datum_size},
+             '$min': {'first_index': (self._datum_count[resource_id] - count)},
+             '$max': {'last_index': self._datum_count[resource_id]-1}},
+            upsert=True)
+
+    def _check_start(self, doc):
+        if self._start_found:
+            raise RuntimeError(
+                "The serializer in suitcase-mongo expects "
+                "documents from one run only. Two `start` documents were "
+                "received.")
+        else:
+            self._start_found = True
+
+    def __repr__(self):
+        # Display connection info in eval-able repr.
+        return "temp repr"  # f'{type(self).__name__}(run_uid={self._run_uid})'
```

Comment:
For now it's fine to remove this method entirely, which would be more informative than return `"temp repr"`.

Reply:
removed